ui/apps/platform/src/Containers/MainPage/InviteUsersModal.tsx

```diff
@@ -0,0 +1,161 @@
+import React, { useEffect, ReactElement } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { createStructuredSelector } from 'reselect';
+import { Modal, ModalVariant, ModalBoxBody, ModalBoxFooter, Button } from '@patternfly/react-core';
+import { useFormik } from 'formik';
+import * as yup from 'yup';
+
+import { selectors } from 'reducers';
+import { actions as inviteActions } from 'reducers/invite';
+import { actions as authActions, types as authActionTypes } from 'reducers/auth';
+import { actions as groupActions } from 'reducers/groups';
+import { actions as roleActions, types as roleActionTypes } from 'reducers/roles';
+import { AuthProvider } from 'services/AuthService/AuthService';
+import { dedupeDelimtedString } from 'utils/textUtils';
+import { mergeGroupsWithAuthProviders } from '../AccessControl/AuthProviders/authProviders.utils';
+import InviteUsersForm from './InviteUsersForm';
+
+type InviteFormValues = {
+    emails: string;
+    provider: string;
+    role: string;
+};
+
+// email validation from discussion in Yup repo,
+// https://github.com/jquense/yup/issues/564#issuecomment-536068508
+const isEmail = (value) => /^[^\s@]+@[^\s@]+\.[^\s@]+$/.test(value);
+const validationSchema = yup.object().shape({
+    emails: yup
+        .string()
+        .transform((value) => dedupeDelimtedString(value).join(',')) // dedupe - optional step
+        .required('At least one email is required')
+        .test('emails', 'Invalid email address', (value) =>
+            Boolean(
+                value &&
+                    value
+                        .split(',')
+                        .map((v) => v.trim())
+                        .every(isEmail)
+            )
+        ), // .string().trim().required('At least one email is required'),
+    provider: yup.string().required('An auth provider is required'),
+    role: yup.string().required('A role is required'),
+});
+
+const defaultInviteFormValues: InviteFormValues = {
+    emails: '',
+    provider: '',
+    role: '',
+};
+
+const feedbackState = createStructuredSelector({
```

Comment:
A question for learning, not a change request: what are pro and con of selectors into Redux store versus ordinary requests in the modal?

Reply:
Good question. We still use the sagas for these values in the Auth Provider section, even though the presentation has been changed to PatternFly, AND before the user even logs in, so we can present them with any available Auth Providers as options in the login box.

Rather than creating a different, redundant way of requesting the auth providers and groups here, I elected to re-use the existing data that already exists in the Redux store.

(At some time in the future, this system could be reworked, but I don't want to increase the scope of changes for this new, minor feature.)